src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -95,6 +95,15 @@ class LinalgBackendEigen : public LinalgBackendBase
 	DEFINE_FOR_ALL_PTYPE(BACKEND_GENERIC_IN_PLACE_ADD, SGMatrix)
 	#undef BACKEND_GENERIC_IN_PLACE_ADD
 
+	/** Implementation of @see LinalgBackendBase::apply */
+	#define BACKEND_GENERIC_IN_PLACE_APPLY(Type, Container) \
+	virtual void apply(SGMatrix<Type>& a, Container<Type>& b, Container<Type>& result) const \
```

Comment:
@karlnapf sry I missed this. I don't think I can overload `dot` method for both mat*vec, and vec*vec in C++ since they have different return types.